.github/workflows/build.yml

```diff
@@ -64,28 +64,14 @@ jobs:
         if: matrix.os == 'macos-13' || matrix.os == 'macos-14'
         run: |
           brew install make clang-format
-      # The build step includes :spotlessJavaBuild.
-      - name: Build Cooja and Documentation
-        uses: gradle/gradle-build-action@v3
-        with:
-          gradle-version: 8.7
-          cache-read-only: false
-          arguments: build -Perrorprone
-          build-root-directory: contiki-ng/tools/cooja
-      # The Jar is important for MSPSim in Contiki-NG.
-      - name: Build Jar
-        uses: gradle/gradle-build-action@v3
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
         with:
-          gradle-version: 8.7
           cache-read-only: false
-          arguments: jar
-          build-root-directory: contiki-ng/tools/cooja
-      - name: Test MSPSim
-        uses: gradle/gradle-build-action@v3
-        with:
-          gradle-version: 8.7
-          arguments: test
-          build-root-directory: contiki-ng/tools/cooja
+      # The build step includes :spotlessJavaBuild.
+      - name: Build and test
```

Comment:
The gradle-build-action did some caching of build artifacts, does this step get the caching from the previous step?

Reply:
As far as I understand, they do the same thing but I have not verified the caching. How are the artifacts used?

> Any workflow that uses gradle/gradle-build-action@v3 will transparently delegate to gradle/actions/setup-gradle@v3. Users are encouraged to update their workflows

Reply:
I had a look in the cache data in the actions and there's a bunch of stuff there from the PR, so it's at least caching something, I assume the cache will get used in the next run then.